lib/config/executor/container_vm.go

```diff
@@ -44,6 +44,21 @@ type Common struct {
 	Notes string `vic:"0.1" scope:"hidden" key:"notes"`
 }
 
+// Common data (specifically for a containerVM) between managed entities, across execution environments.
+type CommonSpecForVM struct {
```

Comment:
This is not really "common" any more since its only used by one struct, `ExecutorConfig`. You can just as well embed this like: 
```go
type ExecutorConfig struct {
  Common struct {
  // A reference to the components hosting execution environment, if any
   	ExecutionEnvironment string
  
 	// Unambiguous ID with meaning in the context of its hosting execution environment
 	ID string `vic:"0.1" scope:"read-only" key:"id"`
 
 	// Convenience field to record a human readable name
 	Name string `vic:"0.1" scope:"hidden" key:"name"`
 
 	// Freeform notes related to the entity
 	Notes string `vic:"0.1" scope:"hidden" key:"notes"`
   } `vic:"0.1" scope:"read-only" key:"common"`
```
At the very least, I think you should rename it to something like `ExecutorConfigCommon`.